xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-api/src/main/java/org/xwiki/index/internal/DefaultTasksManager.java

```diff
@@ -159,6 +148,7 @@ public void initialize()
             MBEAN_NAME);
         this.queue = new PriorityBlockingQueue<>(11, Comparator.comparingLong(TaskData::getTimestamp));
         this.latestTimestampTasksMap = new ConcurrentHashMap<>();
+        startThread();
```

Comment:
The thread is now started during the initialization and the tasks queue when the wikis are ready (and migrated).